plugins/catalog-backend-module-gitlab/src/providers/GitlabDiscoveryEntityProvider.ts

```diff
@@ -177,6 +177,13 @@ export class GitlabDiscoveryEntityProvider implements EntityProvider {
         continue;
       }
 
+      if (
```

Comment:
Could we push this even further down the client making the request and have the request omit forks as an option to the GitLab API instead? Thinking that it's better if we omit them entirely and maybe save on API requests to upstream?

Reply:
This would be the right thing to do but unfortunately [GitLabClient.listProjects](https://github.com/alexmanno/backstage/blob/feat/gitlab-entity-provider-skip-forked-repos/plugins/catalog-backend-module-gitlab/src/lib/client.ts#L67) uses the ["List a group's projects"](https://docs.gitlab.com/ee/api/groups.html#list-a-groups-projects) api and the ["List all projects"](https://docs.gitlab.com/ee/api/projects.html#list-all-projects) api and these apis doesn't have the ability to filter forks...

Reply:
Thinking about it, this might be a good idea for another part of the code, [this archived option](https://github.com/alexmanno/backstage/blob/feat/gitlab-entity-provider-skip-forked-repos/plugins/catalog-backend-module-gitlab/src/providers/GitlabDiscoveryEntityProvider.ts#L176) can be filtered by api! 
If I have a few minutes work on a new PR these days!

Reply:
Hi @benjdlambert, can we consider this change request done?

Reply:
Hey sorry - I missed this notification - yeah OK, that's unfortunate that we can't filter at the request level.

Thanks for checking though!